eastasianwidth.js

```diff
@@ -283,15 +283,28 @@ eaw.length = function(string) {
 };
 
 eaw.slice = function(text, start, end) {
+  textLen = eaw.length(text)
   start = start ? start : 0;
   end = end ? end : 1;
+  if (start < 0) {
+      start = textLen + start;
+  }
+  if (end < 0) {
+      end = textLen + end;
+  }
   var result = '';
+  var eawLen = 0;
   for (var i = 0; i < text.length; i++) {
```

Comment:
I think we should use `stringToArray(text)` to support non-BMP characters.

Reply:
Thanks to pointing out. Yes, we should use `stringToArray(text)` to support surrogate-pair characters.

I fixed it following commit.
https://github.com/komagata/eastasianwidth/pull/6/commits/0f2098de9bdeb4afd5bde03d580a28cbbf06e43b

But on non-BMP characters, current implementation of `eastAsianWidth(character)` seems not handle non-BMP characters (a.k.a Emoji and so on). 
Code points of on-BMP characters are described at http://www.asahi-net.or.jp/~ax2s-kmtn/ref/unicode/emoji.html
We have to add some code point ranges and decide its widths carefully. So supporting non-BMP characters is rather tough work and will makes larger diff containing multiple topic. It will make harder to review code too.

So I suggest making another pull request when we want to support non-BMP characters.

Reply:
The `East_Asian_Width` of emoji was `Neutral` before Unicode 8.0, and is `Wide` in Unicode 9.0 and later. To make `eastAsianWidth(emoji)` be 2, we have to update the data of `eastAsianWidth` to those of newer Unicode Character Database.